find.go

```diff
@@ -53,6 +53,109 @@ func (s *server) findMultihashSubtree(w http.ResponseWriter, r *http.Request) {
 	}
 }
 
+func (s *server) findMetadata(w http.ResponseWriter, r *http.Request) {
+	var rb []byte
+	discardBody(r)
+	if r.Method != http.MethodGet {
+		http.Error(w, "", http.StatusNotFound)
+		return
+	}
+
+	rcode, resp := s.doFindMetadata(r.Context(), r.Method, findMethodOrig, r.URL, rb)
+	if rcode != http.StatusOK {
+		http.Error(w, "", rcode)
+		return
+	}
+	httpserver.WriteJsonResponse(w, http.StatusOK, resp)
+}
+
+func (s *server) doFindMetadata(ctx context.Context, method, source string, req *url.URL, body []byte) (int, []byte) {
```

Comment:
Not much point refactoring this into a separate function considering it is used once and only by find metadata above.